src/Engine/MapController.java

```diff
@@ -450,6 +448,7 @@ private boolean executeGameAction(GameAction action)
       if( events.size() > 0 )
       {
         actionOK = true; // Invalid actions don't produce events.
+        currentActor = myGameInputHandler.getActingUnit();
```

Comment:
Pretty sure this still doesn't handle AIs

Reply:
It should - AI actions still go through `executeGameAction()`.

Reply:
Right, but AIs _don't_ go through `myGameInputHandler`.
We need a way to harvest the acting unit a) from the AI directly, b) from the event, or c) from the animation. (Also, I'm curious why adding a "get acting unit if applicable" doesn't make sense for `GameAnimation`, but it makes sense to put a drawing-logic variable in `Unit`)
Test it. There's 'ghosting'.